storage/InsertDestination.cpp

```diff
@@ -682,8 +685,11 @@ void PartitionAwareInsertDestination::bulkInsertTuplesWithRemappedAttributes(
     // set a bit in the appropriate TupleIdSequence.
     accessor->beginIteration();
     while (accessor->next()) {
-      TypedValue attr_val = accessor->getTypedValue(attribute_map[partition_attribute_id]);
-      partition_membership[partition_scheme_header_->getPartitionId(attr_val)]
+      PartitionSchemeHeader::PartitionValues values;
+      for (const attribute_id attr_id : partition_scheme_header_->getPartitionAttributeIds()) {
+        values.push_back(accessor->getTypedValue(attr_id));
+      }
```

Comment:
Did you take a look at the function ``getTupleWithAttributes()``. It may be useful here. Moreover the ``tuple`` class can compute hash using the ``getTupleHash`` function.